core/registration.c

```diff
@@ -193,8 +194,21 @@ static void prv_handleRegistrationReply(lwm2m_transaction_t * transacP,
     coap_packet_t * packet = (coap_packet_t *)message;
     lwm2m_server_t * targetP = (lwm2m_server_t *)(transacP->userData);
 
-    if (targetP->status == STATE_REG_PENDING)
+    if (targetP->status == STATE_REG_PENDING || targetP->status == STATE_REG_PARTIAL_PROGRESS)
     {
+        if (IS_OPTION(packet, COAP_OPTION_BLOCK1))
+        {
+            uint32_t block1_num;
+            uint8_t  block1_more;
+            uint16_t block1_size;
+            uint32_t block1_offset;
+            coap_get_header_block1(packet, &block1_num, &block1_more, &block1_size, &block1_offset);
+            if (block1_more)
+            {
+                targetP->block_offset = block1_offset + block1_size;
```

Comment:
block1_more just indicates whether or not the final response to the request is included. Any remaining blocks still need to be sent (as long as the response code is 2,xx). The logic isn't right with a block size change either. Suppose block 0 is sent with a size of 1024 and the response comes back with a block size of 512. Changing to the smaller block size means block 1 was already sent with block 0 and the next block sent should be block 2 with a size of 512.

Reply:
block1_size would also need to be saves and used when setting the block1 option in place of MAX_BLOCK1_SIZE.

Reply:
Registration update could return 2.04 changed. To easily check for any success code, check if (packet->code & 0xE0) == (2 << 5)